core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1249,7 +1260,7 @@ class GroupCoordinator(val brokerId: Int,
     // for new members. If the new member is still there, we expect it to retry.
     completeAndScheduleNextExpiration(group, member, NewMemberJoinTimeoutMs)
 
-    maybePrepareRebalance(group, s"Adding new member $memberId with group instance id $groupInstanceId")
+    maybePrepareRebalance(group, s"Adding new member $memberId with group instance id $groupInstanceId. Member joined due to $reason")
```

Comment:
The output log is quite hard to follow at the moment. Example:

```
[2021-12-17 11:29:16,061] INFO [GroupCoordinator 0]: Preparing to rebalance group test in state PreparingRebalance with old generation 1 (__consumer_offsets-48) (reason: Adding new member console-consumer-1d5a9905-c271-4700-a817-62fc9b9f28fc with group instance id None. Member joined due to rebalance failed due to class org.apache.kafka.common.errors.MemberIdRequiredException error: The group member needs to have a valid member id before actually entering a consumer group.) (kafka.coordinator.group.GroupCoordinator)
```

How about doing the following? For each reason, we could add `; client reason: $reason`. With this, we will always have (reason: ....; client reason: ...) in each rebalance logs. It might be clearer. What do you think?

Reply:
+1 for `; client reason: $reason`